crates/pathfinder/src/sequencer.rs

```diff
@@ -1607,6 +1659,58 @@ mod tests {
                 .unwrap();
         }
 
+        #[tokio::test]
+        async fn test_deploy_account() {
+            let (_jh, client) = setup([(
+                "/gateway/add_transaction",
+                response!("0.10.1/deploy_account/response.json"),
+            )]);
+
+            let req =
+                include_str!("../fixtures/sequencer/0.10.1/deploy_account/deploy_account.json");
+            let req: request::add_transaction::AddTransaction =
+                serde_json::from_str(req).expect("Request parsed from JSON");
+            assert_matches!(
+                req,
+                request::add_transaction::AddTransaction::DeployAccount(_)
+            );
+
+            let req = if let request::add_transaction::AddTransaction::DeployAccount(
+                deploy_account,
+            ) = req
+            {
+                deploy_account
+            } else {
+                unreachable!();
+            };
+
+            let res = client
+                .add_deploy_account(
+                    req.version,
+                    req.max_fee,
+                    req.signature,
+                    req.nonce.expect("DEPLOY_ACCOUNT nonce"),
```

Comment:
And this expect() here is also required because of the non-optional nonce argument.